api/v1beta1/common_types.go

```diff
@@ -45,18 +45,18 @@ const (
 	// When removing also remove from staticcheck exclude-rules for SA1019 in golangci.yml.
 	ClusterTopologyManagedFieldsAnnotation = "topology.cluster.x-k8s.io/managed-field-paths"
 
-	// ClusterTopologyMachineDeploymentLabelName is the label set on the generated  MachineDeployment objects
+	// ClusterTopologyMachineDeploymentNameLabel is the label set on the generated  MachineDeployment objects
 	// to track the name of the MachineDeployment topology it represents.
-	ClusterTopologyMachineDeploymentLabelName = "topology.cluster.x-k8s.io/deployment-name"
+	ClusterTopologyMachineDeploymentNameLabel = "topology.cluster.x-k8s.io/deployment-name"
 
 	// ClusterTopologyUnsafeUpdateClassNameAnnotation can be used to disable the webhook check on
 	// update that disallows a pre-existing Cluster to be populated with Topology information and Class.
 	ClusterTopologyUnsafeUpdateClassNameAnnotation = "unsafe.topology.cluster.x-k8s.io/disable-update-class-name-check"
 
-	// ProviderLabelName is the label set on components in the provider manifest.
+	// ProviderNameLabel is the label set on components in the provider manifest.
 	// This label allows to easily identify all the components belonging to a provider; the clusterctl
 	// tool uses this label for implementing provider's lifecycle operations.
-	ProviderLabelName = "cluster.x-k8s.io/provider"
+	ProviderNameLabel = "cluster.x-k8s.io/provider"
```

Comment:
As per @sbueringer this should be 
```suggestion
	ProviderLabel = "cluster.x-k8s.io/provider"
```
(the label is just provider, not provider-name)

Reply:
As per the @sbueringer suggestion in reply https://github.com/kubernetes-sigs/cluster-api/pull/7618#discussion_r1034255049 I made this exception. Please do let me know if I should revert it?

Reply:
I'm fine with `ProviderNameLabel` for the stated reason. @fabriziopandini I'm also fine with `ProviderLabel` if you prefer it.